packages/astro/src/cli/index.ts

```diff
@@ -161,14 +161,16 @@ async function runCommand(cmd: string, flags: yargs.Arguments) {
 	});
 	if (!astroConfig) return;
 	telemetry.record(event.eventCliSession(cmd, userConfig, flags));
+	let tsconfig = loadTSConfig(root);
+	let settings = createSettings(astroConfig, tsconfig?.config, tsconfig?.path);
```

Comment:
Nit: I can already see the day we'll have to refactor `createSettings` to accept an options bag. Maybe worth the refactor now?

Reply:
Done here: https://github.com/withastro/astro/pull/4771/commits/b515843e4188e320d0916473289b243bb5d38aa2